scripts/utils/convert_guidelines.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2023 Baumer (www.baumer.com)
+# SPDX-License-Identifier: Apache-2.0
+
+"""This script converting the Zephyr coding guideline rst file to a output file
+or print the output to the console. Which than can be used by a tool which
+needs to have that information in a specific format (e.g. for cppcheck).
+Or simply use the rule list to generate a filter to suppress all other rules
+used by default from such a tool.
+"""
+
+import sys, getopt, re
+
+def main(argv):
+    inputfile = ''
+    format = ''
+    outputfile = sys.stdout
+    supported_formats = ['cppcheck']
+
+    try:
+        opts, _ = getopt.getopt(argv,"hi:f:o:",["input=","format=",
+                                                 "output=",])
+    except getopt.GetoptError:
+        print('Unsupported option please use one of the following:')
+        print('convert_guidelines.py -i <inputfile> -f <format> [-o <outputfile>]')
+        sys.exit(2)
+    for opt, arg in opts:
+        if opt == "-h":
+            print('convert_guidelines.py -i <inputfile> -f <format> [-o <outputfile>]')
+            print('If outputfile is not specified, print to stdout.')
+            print('Supported formats:' + str(supported_formats))
+            sys.exit()
+        elif opt in ("-i", "--input"):
+            inputfile = arg
+        elif opt in ("-f", "--format"):
+            if arg not in supported_formats:
+                print('Unsupported format please use one of the following:')
+                print(supported_formats)
+                sys.exit(2)
+            format = arg
+        elif opt in ("-o", "--output"):
+            outputfile = open(arg, "w")
+
+    try:
+        file_stream = open(inputfile, 'rt', errors='ignore')
+    except Exception:
+        print('Error opening ' + inputfile +'.')
+        sys.exit()
+
+    # Set the print functions for the format
+    if format == 'cppcheck':
```

Comment:
Wouldn't it be cleaner and simpler to create a `RuleFormatter` class and then derive it with `CppCheckFormatter`?

Reply:
Changed to the suggested handling with classes instead of single functions